### DISCUSSION TITLE
Form based search results button/export

### DISCUSSION THREAD

**Discussion by jkig77** [burkybang/SuiteAdvanced-Feedback#40] [Ideas]
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
In the spirit of implementation and having multiple forms for a record for data importing, it would be helpful to have a button on the record to export to excel all of the exposed header and line fields on any given form at time of button tick. Then we can have a current custom form template export with sample data to provide to the client with the exact fields needed to filled out per custom form. This would be a huge time saver for these larger multi subsidiary projects. To further advance this for customer use, along with it being a static export by pressing a button there could be a unique linking saved search created. Therefore a customer can export mass data based on the exposed fields on a custom form. To add one additional need would be to recognize which fields are mandatory and flag those column names with a asterisk in the saved search and excel export.

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by burkybang** [comment]
This is a really great idea. I can absolutely see how that would be helpful. Thank you!

Here are some notes to help keep me organized while I think about how I could implement this.

- Button: Download CSV Import Template for Form
  - Create a CSV file (not sure if I could do Excel file) with headers of all the body and line fields from the current form
  - Indicate mandatory fields with an *
  - Download CSV file

- Button: Create Saved Search from Form
  - Create a saved search with headers of all the body and line fields from the current form
  - Indicate mandatory fields with an *
  - Open the new saved search in a new tab

**Comment by NicoleAnswer** [comment]
I love this idea too :)  This would truly be a game changer.

**Comment by jkig77** [comment]
@burkybang I see there is an Export Fields button in the SA menu that maybe somewhat related to this request. I tried it on an item master as I'm building templates manually. So far I like the export and can scrub down the data faster. What would be helpful if you can add the following field columns as we see it in the form.

![image](https://github.com/burkybang/SuiteAdvanced-Feedback/assets/101218641/0c7ae427-3ab3-4919-8230-c1ae26760b04)

**Reply by burkybang** [reply]
Can you make an example file just so I understand fully? I'd hate to develop something and then it's not quite what you were thinking. It doesn't need to include every row but at least a few body and sublist fields.
